doc/topics/async-system.md

```diff
@@ -0,0 +1,209 @@
+
+An example will help to illustrate why we need `AsyncSystem` at all.
+
+@mermaid{async-system-load-example}
+
+Let's say we're trying to load a tile in a [3D Tiles](https://github.com/CesiumGS/3d-tiles) tileset. First, we need to start a network request to download the tile's content. We'll do an HTTP GET for a particular URL, and some time tens to hundreds to even thousands of milliseconds later, the tile content at that URL will be downloaded and ready for us to use.
+
+Once it's downloaded, we need to process that tile content that we just downloaded. That means:
+
+* Parse the JSON it contains,
+* Decompress the Draco-encoded meshes or the JPEG-encoded images,
+* Process the triangle meshes for use with a physics engine,
+* And lots more!
+
+We want to do as much of this kind of work as we possibly can in a background thread, because CPUs have multiple cores these days and the main thread is usually busy doing other important work.
+
+Unfortunately, there is inevitably at least a little bit of tile loading work that can't be done in a background thread, though. Game engines, for example, usually have strict rules against creating game objects anywhere other than the main thread. We can't get anything onto the screen without creating game objects, so after the background work is complete, we then need to continue processing this tile in the main thread in order to do the final preparation to render it.
+
+`AsyncSystem` gives us an elegant way to express this kind of sequential process involving a series of asynchronous steps.
+
+## Creating an AsyncSystem {#creating-an-asyncsystem}
+
+Most applications have a single `AsyncSystem` that is used throughout. It is constructed from an [ITaskProcessor](@ref CesiumAsync::ITaskProcessor) instance, which is a simple interface used to perform some work in a background thread. The simplest possible implementation looks like this:
+
+\snippet{trimleft} ExamplesAsyncSystem.cpp simplest-task-processor
+
+This implementation will work well, but it isn't very efficient because a brand new thread is created for each task. Most applications will implement this interface using a thread pool, task graph, or similar functionality that their application already contains.
+
+The `AsyncSystem` can then be created as follows:
+
+\snippet{trimleft} ExamplesAsyncSystem.cpp create-async-system
+
+`AsyncSystem` instances can be safely and efficiently stored and copied by value. This makes it easy to make them available wherever they're needed, including in lambda captures. You can think of an `AsyncSystem` instance as acting a bit like a smart pointer.
+
+\snippet{trimleft} ExamplesAsyncSystem.cpp capture-by-value
+
+However, it is essential that the last `AsyncSystem` instance be destroyed only after everything is done using it. So a common pattern is to create and store it as a static local in an accessor function:
```

Comment:
"using it." Using what? Any instance of the AsyncSystem, or a particular one?

Reply:
I've improved the wording here as well in the part about that previously talked about smart pointers. See what you think.